File: src/mod10_design_pattern/lesson10_flyweight/java/Owl.java
package mod10_design_pattern.lesson10_flyweight.java;

public class Owl implements MagicalCreature {

    private String type;
    private String color;

    public Owl(String type, String color) {
        this.type = type;
        this.color = color;
    }

    @Override
    public void display(String name, String owner) {
        System.out.println("Owl{" +
                "type='" + type + '\'' +
                ", color='" + color + '\'' +
                ", name='" + name + '\'' +
                ", owner='" + owner + '\'' +
                '}');
    }
}

File: src/mod10_design_pattern/lesson10_flyweight/kotlin/Owl.kt
package mod10_design_pattern.lesson10_flyweight.kotlin

class Owl(
    private val type: String,
    private val color: String
) : MagicalCreature {

    override fun display(name: String, owner: String) {
        println("Owl's name is $name, owner is $owner")
    }
}